core/src/program/mod.rs

```diff
@@ -167,14 +167,14 @@ impl Kernel {
 impl Serializable for Kernel {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
         debug_assert!(self.0.len() <= MAX_KERNEL_PROCEDURES);
-        target.write_u16(self.0.len() as u16);
+        target.write_usize(self.0.len());
         target.write_many(&self.0)
     }
 }
 
 impl Deserializable for Kernel {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
-        let len = source.read_u16()?.into();
+        let len = source.read_usize()?;
```

Comment:
We should make sure the `len` does not exceed `MAX_KERNeL_PROCEDURES` here, and return an error if it does.